cmd/kaspawallet/create.go

```diff
@@ -42,30 +44,29 @@ func create(conf *createConfig) error {
 			return errors.Errorf("Public key is too long")
 		}
 
-		publicKey, err := hex.DecodeString(string(line))
-		if err != nil {
-			return err
-		}
-
-		publicKeys = append(publicKeys, publicKey)
+		extendedPublicKeys = append(extendedPublicKeys, string(extendedPublicKey))
 	}
 
-	err = keys.WriteKeysFile(
-		conf.NetParams(), conf.KeysFile, encryptedPrivateKeys, publicKeys, conf.MinimumSignatures, conf.ECDSA)
+	cosignerIndex, err := libkaspawallet.MinimumCosignerIndex(signerExtendedPublicKeys, extendedPublicKeys)
```

Comment:
```
I think you want to sort the public keys outside MinimumCosignerIndex, since the sorting also affects things outside MinimumCosignerIndex
```

The sort is an implementation details, why should the user be aware of it?

```
After reading some more of the code, I begin thinking maybe it could be a good idea to create a type sortedExtendedPublicKeys that will guarantee the sortiness.
```

The assumption is that keys are always ordered except in exported functions

Reply:
> The sort is an implementation detail, why should the user be aware of it?

The sort is not an implementation detail, it is defined in the BIP.

> The assumption is that keys are always ordered except in exported functions
Couldn't it be nice if the assumption had some (even if weak) guarantee using the type-system.

Correct me if I'm wrong, but it seems like addressesToQuery assumes `s.keysFile.ExtendedPublicKeys` is sorted, while in reality it isn't so.

Even if it is, I'd really love it if you had to call sortPublicKeys only once, not every time you need this assumption - this is very error prone, and someone will forget.

Reply:
Right now libkaspawallet is not related to the keys file, so I don't want any assumption about the order of keys in the caller site